lib/set-script.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+const log = require('npmlog')
+const usageUtil = require('./utils/usage.js')
+const { localPrefix } = require('./npm.js')
+const fs = require('fs')
+const usage = usageUtil('set-script', 'npm set-script [<script>] [<command>]')
+const completion = require('./utils/completion/none.js')
+const parseJSON = require('json-parse-even-better-errors')
+const rpj = require('read-package-json-fast')
+
+const cmd = (args, cb) => set(args).then(() => cb()).catch(cb)
+
+const set = async function (args) {
+  if (process.env.npm_lifecycle_event === 'postinstall')
+    log.error('Scripts can\'t set from the postinstall script')
+
+  // Parse arguments
+  if (args.length !== 2) {
+    log.error('set-script', 'Expected 2 arguments: got ' + args.length)
+    return
```

Comment:
similarly, here will *resolve* the overarching async function's promise, when it should probably be rejecting it?